### PR TITLE
Trim token & correct PRs URL

### PR DESCRIPTION
**Pull request by hoegaarden**
"for cases where the token is prvided with leading/trailing whitespaces we
just trim them here. The assumption is that the token itself will never have
whitespaces. If that is not true anymore, this breaks."

/cc @pswica 
/assign @justaugustus

### CODE REVIEW

File: lib/gitlib.sh [hunk 1]
```diff
@@ -21,7 +21,8 @@
 # CONSTANTS
 ###############################################################################
 GITHUB_TOKEN=${FLAGS_github_token:-$GITHUB_TOKEN}
-[[ -n $GITHUB_TOKEN ]] && GITHUB_TOKEN_FLAG=("-u" "$GITHUB_TOKEN:x-oauth-basic")
+GITHUB_TOKEN="$( echo "$GITHUB_TOKEN" | tr -d '[:space:]' )"
```
Comment (justaugustus) [inline lib/gitlib.sh]: So it was an errant space all along?

File: lib/gitlib.sh [hunk 2]
```diff
@@ -21,7 +21,8 @@
...
+[[ -n $GITHUB_TOKEN ]] && GITHUB_TOKEN_FLAG=("-u" "${GITHUB_TOKEN}:x-oauth-basic")
```
Comment (justaugustus) [inline lib/gitlib.sh]:
```suggestion
[[ -n "$GITHUB_TOKEN" ]] && GITHUB_TOKEN_FLAG=("-u" "${GITHUB_TOKEN}:x-oauth-basic")
```

File: lib/gitlib_test.sh [hunk 3]
```diff
@@ -156,4 +158,28 @@ TEST_get_team_members() {
     'get_team_members can handle invalid responses and reports no members'
 }
 
+TEST_pending_prs() {
+  echo "Testing gitlab::pendig_prs"
+  echo
+
+  local branch='release-1.12'
```
Comment (justaugustus) [inline lib/gitlib_test.sh]:
```suggestion
  local branch='release-1.15'
```

File: lib/gitlib_test.sh [hunk 4]
```diff
@@ -156,4 +158,28 @@ TEST_get_team_members() {
     'get_team_members can handle invalid responses and reports no members'
 }
 
+TEST_pending_prs() {
+  echo "Testing gitlab::pendig_prs"
```
Comment (justaugustus) [inline lib/gitlib_test.sh]:
```suggestion
  echo "Testing gitlab::pending_prs"
```

File: lib/gitlib_test.sh [hunk 5]
```diff
@@ -156,4 +158,28 @@ TEST_get_team_members() {
...
+  echo
+
+  local branch='release-1.12'
+  local mockResponse='[{"number":123,"milestone":{"title":"foobar"},"user":{"login":"batman"},"updated_at":"2019-07-03T08:48:34Z","title":"best PR"}]'
```
Comment (justaugustus) [inline lib/gitlib_test.sh]: Niceee. :joy:

File: lib/gitlib.sh (hunk 1 above)
Comment (hoegaarden) [reply lib/gitlib.sh]: my guess: trailing newline https://github.com/kubernetes/release/issues/811

File: lib/gitlib_test.sh (hunk 3 above)
Comment (hoegaarden) [reply lib/gitlib_test.sh]: not relevant for the test

File: lib/gitlib.sh (hunk 2 above)
Comment (hoegaarden) [reply lib/gitlib.sh]: I am not too familiar with `[[` vs `[`, mostly use the later one. But did a quick read and it seems it is fine (but not neccesarily needed) to quote inside `[[`.

File: lib/gitlib.sh (hunk 1 above)
Comment (justaugustus) [reply lib/gitlib.sh]: Really great catch, Hannes!

File: lib/gitlib_test.sh (hunk 3 above)
Comment (justaugustus) [reply lib/gitlib_test.sh]: Not relevant for the test, but from the human perspective, it gives a bit of a signal on when we touched that piece of code w/o having to search history.

**Pull request merged**